### PR TITLE
feature/HTDT-368-add-pagination

### PR DESCRIPTION
**Pull request by evgenybutkevich**

### CODE REVIEW

File: app/components/players/list.hbs [hunk 1]
```diff
@@ -3,6 +3,15 @@
         <th {{on "click" (fn @onHeaderClick 'id')}}>id</th>
         <th {{on "click" (fn @onHeaderClick 'firstName')}}>First name</th>
         <th {{on "click" (fn @onHeaderClick 'lastName')}}>Last name</th>
+        <th>
+            Records on page
+            <PowerSelect
+                @options={{@limitValues}}
+                @selected={{@limit}}
+                @onChange={{@selectLimit}} as |limit|>
```
Comment (van1757) [inline app/components/players/list.hbs]: What do you think about wrapping the closing tag to the new line?IMHO, it's kinda difficult to read in this format

File: app/controllers/players/index.js [hunk 2]
```diff
@@ -34,5 +51,19 @@ export default class IndexController extends Controller {
...
+    @action
+    onPageClick(page) {
+        this.page = page;
+    }
+
+    @action
+    selectLimit(limit) {
+        this.limit = limit;
+
+        this.page = 1;
```
Comment (van1757) [inline app/controllers/players/index.js]: How about moving this logic to function? Something like `resetPage`?

File: app/controllers/players/index.js [hunk 3]
```diff
@@ -6,25 +6,42 @@ const ASCENDING_DIRECTION = 'ASC';
...
 
 export default class IndexController extends Controller {
+    get pageArray() {
+        let array = []
+
+        for (let index = 1; index <= Math.ceil(this.model.meta.count / this.limit); index++) {
+            array.push(index);
+        }
+
+        return array;
```
Comment (Egorrr) [inline app/controllers/players/index.js:12]: I guess we can use smth like this there 
Array(N).fill().map((_, i) => i+1);

File: app/controllers/players/index.js [hunk 4]
```diff
@@ -6,25 +6,42 @@ const ASCENDING_DIRECTION = 'ASC';
...
     @action
-    chooseFilterField(filterField) {
+    selectFilterField(filterField) {
         this.filterField = filterField;
     }
 
     @action
     onFilterClick() {
         this.filterValue = this.value;
+
```
Comment (Egorrr) [inline app/controllers/players/index.js]: no need in this empty line here

File: app/controllers/players/index.js [hunk 5]
```diff
@@ -34,5 +51,19 @@ export default class IndexController extends Controller {
...
+    @action
+    onPageClick(page) {
+        this.page = page;
+    }
+
+    @action
+    selectLimit(limit) {
+        this.limit = limit;
+
```
Comment (Egorrr) [inline app/controllers/players/index.js]: same here with empty line

File: app/templates/players/index.hbs [hunk 6]
```diff
@@ -25,4 +25,23 @@
     <button {{on "click" (fn this.onFilterClick)}}>Filter</button>
 </label>
 
-<Players::List @players={{model}} @onHeaderClick={{this.onHeaderClick}} @sortDirection={{this.sortDirection}} />
+<Players::List
+    @players={{model}}
+
+    @onHeaderClick={{this.onHeaderClick}}
+    @sortDirection={{this.sortDirection}}
+
```
Comment (Egorrr) [inline app/templates/players/index.hbs:34]: no need to leave empty lines here

File: app/templates/players/index.hbs [hunk 7]
```diff
@@ -25,4 +25,23 @@
...
+    @limitValues={{this.limitValues}}
+    @limit={{this.limit}}
+    @selectLimit={{this.selectLimit}}/>
+
+<div class="pages">
+    <p>Pages</p>
+
+    {{#each this.pageArray as |index|}}
+    <button class={{if (eq index this.page) "disabled"}} disabled={{if (eq index this.page) "disabled"}}
```
Comment (Egorrr) [inline app/templates/players/index.hbs]: we can apply style on button:disabled I guess, instead of checking it twice and applying a class

**Review by Egorrr — CHANGES_REQUESTED** [review]

File: app/templates/players/index.hbs (hunk 6 above)
Comment (StanevichIvan) [reply app/templates/players/index.hbs:34]: Add https://prettier.io/ script to the package.json for formatting.

**Review by Egorrr — APPROVED** [review]

**Pull request merged**